rita_client/src/exit_manager/requests.rs

```diff
@@ -271,10 +268,23 @@ pub async fn get_exit_list(exit: IpAddr) -> Result<ExitListV2, RitaClientError>
         Err(e) => return Err(RitaClientError::SendRequestError(e.to_string())),
     };
 
-    let value = response.json().await?;
+    let list: SignedExitServerList = match response {
+        Ok(a) => a,
+        Err(e) => {
+            return Err(RitaClientError::MiscStringError(format!(
+                "Failed to get exit list from exit {:?}",
+                e
+            )));
+        }
+    };
 
-    match decrypt_exit_list(&our_privkey.into(), value, &exit_pubkey.into()) {
-        Err(e) => Err(e.into()),
-        Ok(a) => Ok(a),
+    //verify the signature on the list
+    let key = list.data.contract;
+    let sig = list.clone().signature;
+    match list.data.verify(key, sig) {
+        true => Ok(list),
+        false => Err(RitaClientError::MiscStringError(
```

Comment:
where it's feasible try to make a new error on RitaClientError rather than just using miscstring error it's a catch all that ideally we use sparingly and eventually eliminate. Having a specific error type will make it easier for callers to match on and handle the signature failed error if they ever want to.